kernel/tests/write.rs

```diff
@@ -251,3 +270,377 @@ async fn test_invalid_commit_info() -> Result<(), Box<dyn std::error::Error>> {
     ));
     Ok(())
 }
+
+// check that the timestamps in commit_info and add actions are within 10s of SystemTime::now()
+fn check_action_timestamps<'a>(
+    parsed_commits: impl Iterator<Item = &'a serde_json::Value>,
+) -> Result<(), Box<dyn std::error::Error>> {
+    // check that timestamps are within 10s of SystemTime::now()
+    let now: i64 = std::time::SystemTime::now()
```

Comment:
from the [docs](https://doc.rust-lang.org/std/time/struct.SystemTime.html):
> Distinct from the [Instant](https://doc.rust-lang.org/std/time/struct.Instant.html) type, this time measurement is not monotonic. This means that you can save a file to the file system, then save another file to the file system, and the second file has a SystemTime measurement earlier than the first. In other words, an operation that happens after another operation in real time may have an earlier SystemTime!

So I think you need to make the type such that it can go negative when you subtract below, and then take the absolute value to make sure we're _within_ 10 secs

Reply:
oh nice good catch